2.3 release announcement (#239)
mixxxdj/website: content/news/XXXX-XX-XX-mixxx-2-3-0-release.md

title: Mixxx 2.3.0 released
authors: Jan Holthuis
tags: 2.3, 2.3.0, release announcement
status: draft
comments: yes

After a long waiting time, we are proud to announce the release of Mixxx 2.3.0, [download it!]({filename}/pages/download.md#stable)

> This release brings [hotcue colors & labels]({filename}/news/2020-08-25-new-in-2-3-hotcue-colors.md), import of [Serato](2021-02-08-new-in-2-3-serato-support.md) and [Rekordbox]({filename}/news/2020-07-20-new-in-2-3-rekordbox-support.md) libraries, [more accurate key detection]({filename}/news/2021-04-08-new-in-2-3-keyfinder.md), [intro/outro sections](2020-07-09-intro-outro-sections.md), a revamped LateNight skin, multithreaded analysis, [deck cloning]({filename}/news/2020-11-15-new-in-2-3-deck-clone.md), [track color support]({filename}/news/2020-10-24-new-in-2-3-track-colors.md) and more.

Reviewer: we could link to all the bugs fixed in 2.3: 

https://bugs.launchpad.net/mixxx/+bugs?field.searchtext=&orderby=-importance&field.status%3Alist=FIXCOMMITTED&assignee_option=any&field.assignee=&field.bug_reporter=&field.bug_commenter=&field.subscriber=&field.structural_subscriber=&field.milestone%3Alist=83958&field.tag=&field.tags_combinator=ANY&field.has_cve.used=&field.omit_dupes.used=&field.omit_dupes=on&field.affects_me.used=&field.has_patch.used=&field.has_branches.used=&field.has_branches=on&field.has_no_branches.used=&field.has_no_branches=on&field.has_blueprints.used=&field.has_blueprints=on&field.has_no_blueprints.used=&field.has_no_blueprints=on&search=Search

If we migrate to GH Issues in time, linking to those would also be an option.

Reviewer 2: This link works better for this purpose:
https://launchpad.net/mixxx/+milestone/2.3.0

Reviewer: It also shows all the bugs which didn't get fixed so I didn't want to link that...

Reviewer 2: We will remove the unfixed bugs from the mile stone. There will be overlapping, though.

Reviewer 3: the release announcement should list new, fancy feature and a few important bug fixes only.
all fixed bugs and small tweaks shouldn't be on the front page IMO

Author: Okay, instead of pasting the full changelog I'll just link to it and add the Launchpad link.

Reviewer 3: If this was a list it would be easier to read, and more impressive -- like the [2.2 announcement](https://mixxx.org/news/2018-12-24-mixxx-2-2-released/), but not that detailed probably, or at least we don't need the categories if there are only one or two features in some of them.

Let's collect some 'minor' features / additions, like (browsing [2.3 Done](https://github.com/mixxxdj/mixxx/projects/2#column-8315970) right now)
- Opus encoder added
- NN new controller mappings added
- Preferences and controller GUI polished
- ...

Reviewer 3: maybe also a short 'Behind the scenes' paragraph with all elaborate changes invisible to most users, like
- library improvements
- switch to cmake
- ... ?

Author: @ronso0 Intially I planned to copy the changelog from here: https://github.com/mixxxdj/mixxx/blob/1628fc7cb9c27dc1413568e368c182b32f42aa85/CHANGELOG.md#230-unreleased

It doesn't list each small bugfix either (too many). I don't care if we copy it or link to it.

Reviewer 3: I think we can link to that.
and for the announcement itslef we can use a stripped-down version IMO, which contains stuff that most users may care about.
If we're not _toooo_ much in a hurry I can work on that over the weekend (off the grid, and afk actually.. ; )

Author: Done.

![Screenshot of a Mixxx deck with the intro and outro cues and hotcue editing menu]({static}/images/news/screenshot-2.3-beta-deck.png){: style="max-height: 275px" }

> This has been a rather big release, with lots of useful changes, bug fixes and improvements.

Reviewer 3: ```suggestion
This is a rather big release, with lots of useful changes, bug fixes and improvements.
```
I know, for us it's now like 'case closed', 'done!' -- for users it's the start of something

Almost 100 people contributed a total of 7183 changes.
All in all, a whopping 1 million lines of code were changed since the [2.2.4 release]({filename}/news/2020-06-11-Mixxx-2-2-4-released.md)!

**TODO:** Update the numbers above using `git shortlog -sn release-2.2.4..upstream/2.3 | wc -l`, `git log --oneline --no-merges release-2.2.4..upstream/2.3 -- . ':!lib' | wc -l` and `git diff --shortstat release-2.2.4..upstream/2.3 -- . ':!lib`.

Thanks to all the developers and contributors, we really appreciate it!

## Changelog

**TODO**

## Credits

Many people spent their free time working on Mixxx.
Thanks a lot to all of them:

    $ git shortlog -sn release-2.2.4..upstream/2.3

>     2305  Jan Holthuis

Reviewer 3: and we do have a winner : )
:+1: 

maybe we should mention the numbers stand for commits (just guessing),
not for drunk coffees, hours spent with c++ or how often the person started Mixxx..

Author: I can also check if there is a way to list committers without commit numbers. I just thought it would be a nice idea to list everyone who worked on this release to incentivize contributing :D

Reviewer 2: In general such thing is s good idea.
It "just" forgets the translators and manual authors and other work that does not lead to a countable commit.

Reviewer 3: even though I dislike score boards // peolpe assuming bigger numbers are better...
I'm not questioning the list. It is a nice and simple way it illustrates how much work was done  (if people know what a commit is)
